### PR TITLE
cvcs-13 create asn1 decoder wrapper c function using asn1c

### PR DESCRIPTION
**Pull request by mpalash95**
# PR Details
Create asn1 decoder in C - Develop the C wrapper function
## Description
1. Created c wrapper function to decode  binary-encoded ASN.1 messages using asn1c library

## Related Jira Key

https://usdot-carma.atlassian.net/browse/CVCS-13

## How Has This Been Tested?
1. Compared the decoded string with the output of OSS Nokalva Decoder
2. Added unit test to see if the decoded message is not empty

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

Screenshot of Decoded Message:

### CODE REVIEW

File: fedgov-cv-lib-asn1c/src/main/java/decoder_wrapper.c [hunk 1]
```diff
@@ -23,29 +23,44 @@
 JNIEXPORT jstring JNICALL Java_gov_usdot_cv_asn1decoder_Decoder_decodeMsg(JNIEnv *env, jobject obj, jbyteArray encoded_msg)
 {
 	// TODO temporary
-	const char *resultStr = "Decoded Message";
+	const char *resultStr;
```
Comment (saina-ramyar) [inline fedgov-cv-lib-asn1c/src/main/java/decoder_wrapper.c:26]: now with this change, the TODO comment above can be removed

File: fedgov-cv-lib-asn1c/src/main/java/decoder_wrapper.c [hunk 2]
```diff
@@ -23,29 +23,44 @@
...
 
-	// if(rval.code == RC_OK) {
-	// 	}
-	// else {
-	// 	// return error for failure to decode
-	// }
+		asn_fprint(stdout, &asn_DEF_MessageFrame, message);
 
+		char outputBuffer[4096]; // Ensure it's large enough
+		FILE *stream = fmemopen(outputBuffer, sizeof(outputBuffer), "w");
```
Comment (saina-ramyar) [inline fedgov-cv-lib-asn1c/src/main/java/decoder_wrapper.c:46]: add a comment for this line

File: fedgov-cv-asn1decoder/src/test/java/gov/usdot/cv/asn1decoder/ASN1DecoderTest.java [hunk 3]
```diff
@@ -46,16 +49,10 @@ public void ASN1DecoderTest() {
...
-        long end = System.currentTimeMillis();
-
-        System.out.printf("res: %s", mockMsg.getType());
-
-        String expected = "Decoded Message";
-
+        logger.info("Type: %s\n", mockMsg.getType());
+        String decodedMessage= decoder.decode(mockMsg);
+        //Test passes if decoded message has a length greater than 0
+        Assert.assertTrue("Decoded result should not be empty",!decodedMessage.isEmpty());
```
Comment (saina-ramyar) [inline fedgov-cv-asn1decoder/src/test/java/gov/usdot/cv/asn1decoder/ASN1DecoderTest.java:55]: please also a test case where the message is invalid, and an empty string is the output.

File: fedgov-cv-lib-asn1c/src/main/java/decoder_wrapper.c [hunk 4]
```diff
@@ -23,29 +23,44 @@
...
+		if (stream) {
```
Comment (saina-ramyar) [inline fedgov-cv-lib-asn1c/src/main/java/decoder_wrapper.c:47]: please add comments for this "if" statement, and the following lines

File: fedgov-cv-lib-asn1c/src/main/java/decoder_wrapper.c [hunk 5]
```diff
@@ -23,29 +23,44 @@
...
+			asn_fprint(stream, &asn_DEF_MessageFrame, message);
+			fclose(stream);
+			resultStr = strdup(outputBuffer); // Dumping the outputBuffer
+		} else {
+			resultStr = "Failed to allocate memory for output";
+		}
 
+	}
+	else if(rval.code == RC_WMORE) {
```
Comment (saina-ramyar) [inline fedgov-cv-lib-asn1c/src/main/java/decoder_wrapper.c:56]: what does it mean? what kind of data?

File: fedgov-cv-lib-asn1c/src/main/java/decoder_wrapper.c (hunk 2 above)
Comment (mpalash95) [reply fedgov-cv-lib-asn1c/src/main/java/decoder_wrapper.c:47]: Added comments

File: fedgov-cv-asn1decoder/src/test/java/gov/usdot/cv/asn1decoder/ASN1DecoderTest.java (hunk 3 above)
Comment (mpalash95) [reply fedgov-cv-asn1decoder/src/test/java/gov/usdot/cv/asn1decoder/ASN1DecoderTest.java:57]: Added test case for empty string

File: fedgov-cv-lib-asn1c/src/main/java/decoder_wrapper.c (hunk 4 above)
Comment (mpalash95) [reply fedgov-cv-lib-asn1c/src/main/java/decoder_wrapper.c:49]: Added comments for if case

File: fedgov-cv-lib-asn1c/src/main/java/decoder_wrapper.c (hunk 5 above)
Comment (mpalash95) [reply fedgov-cv-lib-asn1c/src/main/java/decoder_wrapper.c:60]: Added more details
